Microsoft.Toolkit.Uwp.UI.Controls/AdvancedCollectionViewSource/AdvancedCollectionViewSource.cs

```diff
@@ -0,0 +1,560 @@
+            }
+
+            _sourceList.Clear();
+        }
+
+        public bool Contains(object item) => _view.Contains(item);
+
+        public void CopyTo(object[] array, int arrayIndex) => _view.CopyTo(array, arrayIndex);
+
+        public bool Remove(object item)
+        {
+            if (IsReadOnly)
+            {
+                throw new Exception("The source is in read-only mode!");
+            }
+
+            _sourceList.Remove(item);
+            return true;
+        }
+
+        public int Count => _view.Count;
+
+        public bool IsReadOnly => _sourceList == null || _sourceList.IsReadOnly;
+
+        public int IndexOf(object item) => _view.IndexOf(item);
+
+        public void Insert(int index, object item)
+        {
+            if (IsReadOnly)
+            {
+                throw new Exception("The source is in read-only mode!");
+            }
+
+            if (_sortDescriptions.Count > 0 || _filter != null)
+            {
+                // no sense in inserting w/ filters or sorts, just add it
+                _sourceList.Add(item);
+            }
+
+            _sourceList.Insert(index, item);
+        }
+
+        /// <summary>
+        /// Removes the <see cref="T:System.Collections.Generic.IList`1"/> item at the specified index.
+        /// </summary>
+        /// <param name="index">The zero-based index of the item to remove.</param><exception cref="T:System.ArgumentOutOfRangeException"><paramref name="index"/> is not a valid index in the <see cref="T:System.Collections.Generic.IList`1"/>.</exception><exception cref="T:System.NotSupportedException">The <see cref="T:System.Collections.Generic.IList`1"/> is read-only.</exception>
+        public void RemoveAt(int index) => Remove(_view[index]);
+
+        /// <summary>
+        /// Gets or sets the element at the specified index.
+        /// </summary>
+        /// <returns>
+        /// The element at the specified index.
+        /// </returns>
+        /// <param name="index">The zero-based index of the element to get or set.</param><exception cref="T:System.ArgumentOutOfRangeException"><paramref name="index"/> is not a valid index in the <see cref="T:System.Collections.Generic.IList`1"/>.</exception><exception cref="T:System.NotSupportedException">The property is set and the <see cref="T:System.Collections.Generic.IList`1"/> is read-only.</exception>
+        public object this[int index]
+        {
+            get { return _view[index]; }
+            set { _view[index] = value; }
+        }
+
+        /// <summary>
+        /// Occurs when the vector changes.
+        /// </summary>
+        public event VectorChangedEventHandler<object> VectorChanged;
+
+        /// <summary>
+        /// Move current index to item
+        /// </summary>
+        /// <param name="item">item</param>
+        /// <returns>success of operation</returns>
+        public bool MoveCurrentTo(object item) => item == CurrentItem || MoveCurrentToIndex(IndexOf(item));
```

Comment:
Why have this when you already have the CurrentItem property that is get and set?